src/AstTranslator.cpp

```diff
@@ -1381,7 +1392,7 @@ std::unique_ptr<RamStatement> AstTranslator::makeNegationSubproofSubroutine(cons
             size_t auxiliaryArity = auxArityAnalysis->getArity(atom);
             // get a RamRelationReference
             auto relRef = translateRelation(atom);
-
+            auto atomArgs = atom->getArguments();
```

Comment:
This should be moved to just before the for loop as the atom is modified on line 1401

Reply:
After moving this line, still causing compile tests fail on all provenance tests.